Lib/_pydecimal.py

```diff
@@ -6076,8 +6076,13 @@ def _convert_for_comparison(self, other, equality_op=False):
     \Z
 """, re.VERBOSE | re.IGNORECASE).match
 
-_all_zeros = re.compile('0*$').match
-_exact_half = re.compile('50*$').match
+# Checks for regex 0*$
+def _all_zeros(d_int,prec=0):
+    return '0' in d_int and d_int.endswith((len(d_int) - prec) * '0')
+
+# Checks for regex 50*$ 
+def _exact_half(d_int, prec=0):
+    return len(d_int) >= prec + 1 and d_int[prec] == '5' and d_int.endswith(max(len(d_int) - prec - 1, 0) * '0')
```

Comment:
```suggestion
    return len(d_int) >= prec + 1 and d_int[prec] == '5' and d_int.endswith( (len(d_int) - prec - 1) * '0')
```
Since `-2 * '0'` is `''`

Reply:
Had not checked if that was the case